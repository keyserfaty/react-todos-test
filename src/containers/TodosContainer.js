import React from 'react'
import Todos from '../components/Todos'

class TodosContainerComponent extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      todos: [],
    };

    this.onAddTodoClick = this.onAddTodoClick.bind(this);
  }

  onAddTodoClick(todo) {
    this.setState({
      todos: [
        ...this.state.todos,
        todo,
      ],
    });
  }

  render() {
    const { onAddTodoClick } = this;
    const { todos } = this.state;

    return (
      <Todos todos={todos} onAddTodoClick={onAddTodoClick} />
    );
  }
}

export default TodosContainerComponent
