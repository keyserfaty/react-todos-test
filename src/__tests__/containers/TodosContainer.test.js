import React from 'react'
import renderer from 'react-test-renderer'

import TodosContainer from '../../containers/TodosContainer'

it('should render a todos input component', () => {
  const tree = renderer.create(
    <TodosContainer />
  ).toJSON();

  expect(tree).toMatchSnapshot()
});
