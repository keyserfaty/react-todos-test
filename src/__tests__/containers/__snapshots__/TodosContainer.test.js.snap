exports[`test should render a todos input component 1`] = `
<div
  id="todos-container">
  <div
    id="todos-input">
    <input
      onChange={[Function]}
      type="text"
      value="" />
    <button
      onClick={[Function]}>
      Add todo!
    </button>
  </div>
  <div
    id="todos-list">
    <ul />
  </div>
</div>
`;
